hermes-management/src/main/java/pl/allegro/tech/hermes/management/domain/dc/MultiDcRepositoryCommandExecutor.java

```diff
@@ -0,0 +1,76 @@
+package pl.allegro.tech.hermes.management.domain.dc;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.common.exception.HermesException;
+import pl.allegro.tech.hermes.common.exception.InternalProcessingException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class MultiDcRepositoryCommandExecutor {
+
+    private static final Logger logger = LoggerFactory.getLogger(MultiDcRepositoryCommandExecutor.class);
+
+    private final RepositoryManager repositoryManager;
+    private final boolean rollbackEnabled;
+
+    public MultiDcRepositoryCommandExecutor(RepositoryManager repositoryManager, boolean rollbackEnabled) {
+        this.repositoryManager = repositoryManager;
+        this.rollbackEnabled = rollbackEnabled;
+    }
+
+    public <T> void execute(RepositoryCommand<T> command) {
+        if (rollbackEnabled) {
+            backup(command);
+        }
+
+        List<DcBoundRepositoryHolder<T>> repositories = repositoryManager.getRepositories(command.getRepositoryType());
+        List<DcBoundRepositoryHolder<T>> succeededRepositories = new ArrayList<>();
+
+        for (DcBoundRepositoryHolder<T> repository : repositories) {
+            try {
+                command.execute(repository.getRepository());
+                succeededRepositories.add(repository);
+            } catch (Exception e) {
+                if (rollbackEnabled) {
+                    rollback(succeededRepositories, command);
```

Comment:
Right now rollback is done for all the repositories (loop inside a `rollback` func). Should we make per repo - dunno :(

Reply:
I mean, we pass `succeededRepositories` here, and then for each of them we call `rollback()`. The actual exception was thrown because of a failure related to the "current" `repository`, for which we don't call `rollback()`.